extensions/cornerstone/src/tools/ImageOverlayViewerTool.tsx

```diff
@@ -27,8 +29,26 @@ interface CachedStat {
  */
 class ImageOverlayViewerTool extends AnnotationDisplayTool {
   static toolName = 'ImageOverlayViewer';
-  private _cachedOverlayMetadata: Map<string, any[]> = new Map();
-  private _cachedStats: { [key: string]: CachedStat } = {};
+  /** Adds the metadata for overlayPlaneModule */
+  public static add(imageId, metadata) {
+    if (_cachedOverlayMetadata.get(imageId) === metadata) {
+      // This is a no-op here as the tool re-caches the data
+      return;
+    }
+    _cachedOverlayMetadata.set(imageId, metadata);
+    delete _cachedStats[imageId];
+  }
+
+  /** Standard getter for metadata */
+  public static get(type: string, query: string | string[]) {
+    if (Array.isArray(query)) {
+      return;
+    }
+    if (type !== 'overlayPlaneModule') {
+      return;
+    }
+    return _cachedOverlayMetadata.get(query);
+  }
```

Comment:
can you separate this out into another file?

Reply:
Should it get separated out?  The add method ensures that cached derived data gets cleared as well, and the derived data is generated by this class.  I can add it separately, just if I do, then I will add a separate method for the _cachedStats data.

Reply:
Having it separated out would match the rest of our designs IMO